homeassistant/components/sql/config_flow.py

```diff
@@ -155,6 +155,7 @@ async def async_step_init(
             db_url = user_input.get(CONF_DB_URL, db_url_default)
```

Comment:
```suggestion
            db_url = self.entry.options.get(CONF_DB_URL, db_url_default)
```

Reply:
db_url should be read from the user input or default url if input missing (as it's optional).
That change you propose would make it impossible to change the url if wanted.

Reply:
But if you change it in the options flow, then the next time you open the options flow it gets reset

Reply:
Not sure what you mean by reset but the schema has the entry options as suggested value so it will propose whatever the user saved last time.